### ISSUE TITLE
Tag runners with OS image

### ISSUE THREAD

**Issue by rootfs** [sustainable-computing-io/aws_ec2_self_hosted_runner#14]
Current the runner tag consists only x86 and linux. Since we support both RHEL and ubuntu, we need to tag the runners correctly so when there are multiple runners, tests can find runners correctly.

Tests:
```yaml
runs-on: [self-hosted, linux, x64]
```

Runner tagging is [here](https://github.com/sustainable-computing-io/aws_ec2_self_hosted_runner/blob/main/entrypoint.sh#L138). Per github [doc](https://docs.github.com/en/actions/hosting-your-own-runners/managing-self-hosted-runners/using-labels-with-self-hosted-runners), we can add labels in this way:

```bash
./config.sh --url <REPOSITORY_URL> --token <REGISTRATION_TOKEN> --labels gpu,x64,linux
```

**Comment by rootfs** [comment]
Once this issue is resolved, we release 0.4

**Comment by SamYuan1990** [comment]
I got the point which using a label to tag self hosted runners by code as https://docs.github.com/en/actions/hosting-your-own-runners/managing-self-hosted-runners/using-labels-with-self-hosted-runners 

and if we just https://docs.github.com/en/actions/using-workflows/workflow-syntax-for-github-actions#jobsjob_idruns-on as
```
runs-on: [self-hosted, linux, x64]
```
which means when we running in parallel... all those three runners has the same label.
Hence, @rootfs , is there any hash specific value can we use for label an instance?
maybe instance id, so that when the jobs running in parallel, we can relay on hash value to test on specific runner.
`steps.create-runner.outputs.instance_id`
